superset/models/core.py

```diff
@@ -180,14 +187,14 @@ def __repr__(self):
         return self.slice_name or str(self.id)
 
     @property
-    def cls_model(self):
+    def cls_model(self) -> Type["BaseDatasource"]:
         return ConnectorRegistry.sources[self.datasource_type]
 
     @property
-    def datasource(self):
+    def datasource(self) -> "BaseDatasource":
         return self.get_datasource
 
-    def clone(self):
+    def clone(self) -> "Slice":
```

Comment:
can explain what this string hints mean?

Reply:
Since this is a method of the `Slice` class, the `Slice` class hasn't been defined yet. Quotes allow us to reference it anyway for the static typechecker to do its thing.

Reply:
sorry, I didn't see this usage on any typing hints reference. Can you give the source?

Reply:
https://www.python.org/dev/peps/pep-0484/#forward-references